docs/authentication.md

```diff
@@ -0,0 +1,43 @@
+# Authentication
+
+## Devise
+
+[Devise](https://github.com/heartcombo/devise) is a flexible authentication solution that provides a full-featured authentication system that provides a complete MVC solution based on Rails engines. Some of the modules it includes:
+
+**Database Authenticatable**, **Registerable**, **Confirmable**, **Timeoutable**, **Omniauthable**..etc
+
+Here is an example of how Devise is used in our User model:
+
+```
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable, :confirmable, :trackable, :omniauthable
+end
+```
+
+Devise is customizable, allowing you to add or remove modules that best suit your application. Devise provides many helper methods like `user_signed_in?` that checks if a user is signed in or `current_user` that returns the **User** record of the signed in user.
+
+### Configuration
+
+In the provided codebase, devise in configured in `config/initializers/devise.rb`.
+
+1. If using the Rails mailer, don't forget to set the value of the constant `DEFAULT_FROM_EMAIL` for your application.
+
+2. If modifying anything related to the sign up or confirmation process, refer to the controllers `RegistrationsController` and `ConfirmationsController` which override and/or inherits from their original devise controllers.
+
+## Omniauth
+
+Google OmniAuth is used to authenticate users via Google's OAuth 2.0 API. This means users can sign in to your application using their Google account credentials. This simplifies the sign-in process for users, as they don't need to remember another username/password.
```

Comment:
can simplify this to say that you can sign in with google and how to set that up. 

also mention how they would disable connecting with google if they dont want this. this would be just to hide the button I guess and not set the keys.